### PR TITLE
[rte:archives] Created extractor #8525

### PR DESCRIPTION
**Pull request by Nick-Triller**
Solves #8525. Flake8 reports no problems.

### CODE REVIEW

File: youtube_dl/extractor/rte.py [hunk 1]
```diff
@@ -11,6 +11,58 @@
...
+            'duration': 255.001,
+        },
+        'params': {
+            'skip_download': 'f4m fails with --test atm'
+        }
+    }
+
+    def _real_extract(self, url):
+        # meta tag with name="av" contains this id, too
+        id_pattern = r'//www.rte.ie/bosco/components/player/iframe.html\?clipid=(\d+)'
```
Comment (dstftw) [inline youtube_dl/extractor/rte.py]: Pattern should include `<iframe` part.

File: youtube_dl/extractor/rte.py [hunk 2]
```diff
@@ -11,6 +11,58 @@
...
+        video_id = None
```
Comment (dstftw) [inline youtube_dl/extractor/rte.py]: Initial `video_id` or `display_id` should be extracted from input URL.

File: youtube_dl/extractor/rte.py [hunk 3]
```diff
@@ -11,6 +11,58 @@
...
+            r'<meta name="thumbnail" content="uri:irus:(.*?)"/?>', webpage, 'thumbnail')
+        thumbnail = 'http://img.rasset.ie/' + thumbnail_id + '.jpg'
+
+        feeds_url = 'http://www.rte.ie/rteavgen/getplaylist/?format=json&id=' + video_id
+        json_string = self._download_json(feeds_url, video_id)
+
+        # f4m_url = server + relative_url
+        f4m_url = json_string['shows'][0]['media:group'][0]['hds_server'] + json_string['shows'][0]['media:group'][0]['hds_url']
+        f4m_formats = self._extract_f4m_formats(f4m_url, video_id)
+        duration = json_string['shows'][0]['media:group'][0]['duration']
```
Comment (dstftw) [inline youtube_dl/extractor/rte.py]: Extraction should be tolerate to missing fields.

File: youtube_dl/extractor/rte.py [hunk 4]
```diff
@@ -11,6 +11,58 @@
...
+        duration = duration / 1000.0
```
Comment (dstftw) [inline youtube_dl/extractor/rte.py]: Use `int_or_none`.

**Comment by Nick-Triller** [comment]
The issues will be resolved on thursday.

**Comment by Nick-Triller** [comment]
All issues were resolved. Note: The empty "Merge" commit is an artifact of a rebase attempt.

**Comment by dstftw** [comment]
Can you squash commits?

**Comment by Nick-Triller** [comment]
I squashed the commits. Everything ready to go! :relaxed:

**Comment by yan12125** [comment]
Since #8497 landed, please move changes in ```youtube_dl/extractors/__init__.py``` to ```youtube_dl/extractors/extractors.py```. Check step 5 of the new [developer instructions](https://github.com/rg3/youtube-dl/blob/master/README.md#adding-support-for-a-new-site) for more information.

**Comment by Nick-Triller** [comment]
I am going to create a new pull request for this change

**Pull request closed without merging**